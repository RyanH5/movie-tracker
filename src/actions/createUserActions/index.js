export const userCreateSuccess = (id) => ({
  type: 'CREATE_USER_SUCCESS',
  id
});

export const requiredIncomplete = (bool) => ({
  type: 'INCOMPLETE_ENTRIES',
  fieldsIncomplete: bool
});

export const creationDenied = (bool) => ({
  type: 'EMAIL_NOT_AVAILABLE',
  emailBeingUsed: bool
});

export const createNewUser = (name, email, password) => {
  return async (dispatch) => {
    dispatch(requiredIncomplete(false));
    dispatch(creationDenied(false));
    if (!name || !email || !password) {
      return dispatch(requiredIncomplete(true));
    }
    try {
      const url = 'http://localhost:3000/api/users/new';
      const response = await fetch(url, {
        method: 'POST',
        body: JSON.stringify(
          {
            name,
            email,
            password
          }
        ),
        headers: {
          'Content-Type': 'application/json'
        }
      });
      const userCreated = await response.json();
      dispatch(userCreateSuccess(userCreated.id));
    } catch (e) {
      dispatch(creationDenied(true));
    }
  };
};
