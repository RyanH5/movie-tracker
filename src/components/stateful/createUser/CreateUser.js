import React, { Component } from 'react';
import { createNewUser, clearForm } from '../../../actions/createUserActions';
import { connect } from 'react-redux';
import './CreateUser.css';

class CreateUser extends Component {
  constructor () {
    super();
    this.state = {
      name: '',
      password: '',
      email: ''
    };
  }

  handleChange = (event) => {
    const {name, value} = event.target;
    this.setState({[name]: value});
  };

  handleSubmit = (event) => {
    event.preventDefault();
    // dispatch an action
    // state updates to new key clearAlerts: true
    // this.props.clearAlerts === true
    //
    const {name, email, password} = this.state;
    this.props.createNewUser(name, email, password);
  };

  render () {
    return (
      <div className="create-user-container d-flex">
        <div className="create-user-form-wrapper">
          <form onSubmit={this.handleSubmit}>
            <span className="">
						Create User
					  </span>
            {
              this.props.invalidForm &&
              <p>Please complete required fields.</p>
            }
            {
              this.props.createAccountFailed &&
              <p>Email is already being used.</p>
            }
            <div className="wrap-input top">
              <input
                type="text"
                name="name"
                value={this.state.name}
                placeholder="ex. John Smith"
                onChange={this.handleChange}
              />
            </div>
            <div className="wrap-input">
              <input
                type="text"
                name="email"
                value={this.state.email}
                placeholder="ex. Jsmith@gmail.com"
                onChange={this.handleChange}
              /></div>
            <div className="wrap-input bottom">
              <input
                type="text"
                name="password"
                value={this.state.password}
                placeholder="PASSWORD"
                onChange={this.handleChange}
              />
            </div>
            <button>SIGN-UP</button>
          </form>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  newUserId: state.newUserId,
  invalidForm: state.invalidForm,
  createAccountFailed: state.createAccountFailed
});

const mapDispatchToProps = (dispatch) => ({
  createNewUser: (name, email, password) => dispatch(createNewUser(name, email, password))
});

export default connect(mapStateToProps, mapDispatchToProps)(CreateUser);
