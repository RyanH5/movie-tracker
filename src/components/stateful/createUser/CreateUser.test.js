import React from 'react';
import { shallow, mount} from 'enzyme';
import { CreateUser } from './CreateUser';
import * as actions from '../../../actions/createUserActions/index';
import { Provider } from 'react-redux';

describe('CreateUser', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CreateUser />);
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should have an initial state', () => {
    expect(wrapper.state().name).toBe('');
    expect(wrapper.state().password).toBe('');
    expect(wrapper.state().email).toEqual('');
  });
});
