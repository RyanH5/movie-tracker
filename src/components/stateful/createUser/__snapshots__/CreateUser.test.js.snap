// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUser should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateUser />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="create-user-form-wrapper"
>
        <form
                onSubmit={[Function]}
        >
                <span
                        className=""
                >
                        Create User
                </span>
                <div
                        className="wrap-input top"
                >
                        <input
                                name="name"
                                onChange={[Function]}
                                placeholder="ex. John Smith"
                                type="text"
                                value=""
                        />
                </div>
                <div
                        className="wrap-input"
                >
                        <input
                                name="email"
                                onChange={[Function]}
                                placeholder="ex. Jsmith@gmail.com"
                                type="text"
                                value=""
                        />
                </div>
                <div
                        className="wrap-input bottom"
                >
                        <input
                                name="password"
                                onChange={[Function]}
                                placeholder="PASSWORD"
                                type="text"
                                value=""
                        />
                </div>
                <button>
                        SIGN-UP
                </button>
        </form>
</div>,
      "className": "create-user-container d-flex",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
>
          <span
                    className=""
          >
                    Create User
          </span>
          <div
                    className="wrap-input top"
          >
                    <input
                              name="name"
                              onChange={[Function]}
                              placeholder="ex. John Smith"
                              type="text"
                              value=""
                    />
          </div>
          <div
                    className="wrap-input"
          >
                    <input
                              name="email"
                              onChange={[Function]}
                              placeholder="ex. Jsmith@gmail.com"
                              type="text"
                              value=""
                    />
          </div>
          <div
                    className="wrap-input bottom"
          >
                    <input
                              name="password"
                              onChange={[Function]}
                              placeholder="PASSWORD"
                              type="text"
                              value=""
                    />
          </div>
          <button>
                    SIGN-UP
          </button>
</form>,
        "className": "create-user-form-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className=""
>
              Create User
</span>,
            undefined,
            undefined,
            <div
              className="wrap-input top"
>
              <input
                            name="name"
                            onChange={[Function]}
                            placeholder="ex. John Smith"
                            type="text"
                            value=""
              />
</div>,
            <div
              className="wrap-input"
>
              <input
                            name="email"
                            onChange={[Function]}
                            placeholder="ex. Jsmith@gmail.com"
                            type="text"
                            value=""
              />
</div>,
            <div
              className="wrap-input bottom"
>
              <input
                            name="password"
                            onChange={[Function]}
                            placeholder="PASSWORD"
                            type="text"
                            value=""
              />
</div>,
            <button>
              SIGN-UP
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create User",
              "className": "",
            },
            "ref": null,
            "rendered": "Create User",
            "type": "span",
          },
          undefined,
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                name="name"
                onChange={[Function]}
                placeholder="ex. John Smith"
                type="text"
                value=""
/>,
              "className": "wrap-input top",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "name",
                "onChange": [Function],
                "placeholder": "ex. John Smith",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                name="email"
                onChange={[Function]}
                placeholder="ex. Jsmith@gmail.com"
                type="text"
                value=""
/>,
              "className": "wrap-input",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "placeholder": "ex. Jsmith@gmail.com",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                name="password"
                onChange={[Function]}
                placeholder="PASSWORD"
                type="text"
                value=""
/>,
              "className": "wrap-input bottom",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "placeholder": "PASSWORD",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SIGN-UP",
            },
            "ref": null,
            "rendered": "SIGN-UP",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="create-user-form-wrapper"
>
          <form
                    onSubmit={[Function]}
          >
                    <span
                              className=""
                    >
                              Create User
                    </span>
                    <div
                              className="wrap-input top"
                    >
                              <input
                                        name="name"
                                        onChange={[Function]}
                                        placeholder="ex. John Smith"
                                        type="text"
                                        value=""
                              />
                    </div>
                    <div
                              className="wrap-input"
                    >
                              <input
                                        name="email"
                                        onChange={[Function]}
                                        placeholder="ex. Jsmith@gmail.com"
                                        type="text"
                                        value=""
                              />
                    </div>
                    <div
                              className="wrap-input bottom"
                    >
                              <input
                                        name="password"
                                        onChange={[Function]}
                                        placeholder="PASSWORD"
                                        type="text"
                                        value=""
                              />
                    </div>
                    <button>
                              SIGN-UP
                    </button>
          </form>
</div>,
        "className": "create-user-container d-flex",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            onSubmit={[Function]}
>
            <span
                        className=""
            >
                        Create User
            </span>
            <div
                        className="wrap-input top"
            >
                        <input
                                    name="name"
                                    onChange={[Function]}
                                    placeholder="ex. John Smith"
                                    type="text"
                                    value=""
                        />
            </div>
            <div
                        className="wrap-input"
            >
                        <input
                                    name="email"
                                    onChange={[Function]}
                                    placeholder="ex. Jsmith@gmail.com"
                                    type="text"
                                    value=""
                        />
            </div>
            <div
                        className="wrap-input bottom"
            >
                        <input
                                    name="password"
                                    onChange={[Function]}
                                    placeholder="PASSWORD"
                                    type="text"
                                    value=""
                        />
            </div>
            <button>
                        SIGN-UP
            </button>
</form>,
          "className": "create-user-form-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className=""
>
                Create User
</span>,
              undefined,
              undefined,
              <div
                className="wrap-input top"
>
                <input
                                name="name"
                                onChange={[Function]}
                                placeholder="ex. John Smith"
                                type="text"
                                value=""
                />
</div>,
              <div
                className="wrap-input"
>
                <input
                                name="email"
                                onChange={[Function]}
                                placeholder="ex. Jsmith@gmail.com"
                                type="text"
                                value=""
                />
</div>,
              <div
                className="wrap-input bottom"
>
                <input
                                name="password"
                                onChange={[Function]}
                                placeholder="PASSWORD"
                                type="text"
                                value=""
                />
</div>,
              <button>
                SIGN-UP
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Create User",
                "className": "",
              },
              "ref": null,
              "rendered": "Create User",
              "type": "span",
            },
            undefined,
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  name="name"
                  onChange={[Function]}
                  placeholder="ex. John Smith"
                  type="text"
                  value=""
/>,
                "className": "wrap-input top",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "name",
                  "onChange": [Function],
                  "placeholder": "ex. John Smith",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  name="email"
                  onChange={[Function]}
                  placeholder="ex. Jsmith@gmail.com"
                  type="text"
                  value=""
/>,
                "className": "wrap-input",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "ex. Jsmith@gmail.com",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  name="password"
                  onChange={[Function]}
                  placeholder="PASSWORD"
                  type="text"
                  value=""
/>,
                "className": "wrap-input bottom",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "PASSWORD",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "SIGN-UP",
              },
              "ref": null,
              "rendered": "SIGN-UP",
              "type": "button",
            },
          ],
          "type": "form",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
